### DISCUSSION TITLE
Obtaining Consent with the User Messaging Platform

### DISCUSSION THREAD

**Discussion by jeremyhalin** [invertase/react-native-firebase#5907] [Q&A] [answered]
On AdMob website, I can create and edit a consent form for GPDR.
I would like to implement it in my React Native app, Google has a [documentation](https://developers.google.com/admob/ump/android/quick-start) about it.
Is it possible to achieve that with `react-native-firebase`? I was previously using the `admob` package of `react-native-firebase` but it's deprecated, and I've been using another lib to manage ads.

There is [this package](https://github.com/birgernass/react-native-ad-consent ), but it doesn't seem to be up to date nor maintain.

**Comment by mikehardy** [comment, marked as answer]
Hi there! With apologies, as you mention the admob package is no longer in this repository, therefore admob questions are simply out of scope. We do have plans to get the code that was here in shape and released separately but it's behind some other projects unfortunately and has been stuck at "really close" for far too long #5329 
That said, the old documentation exists, and I believe it was possible.
https://github.com/invertase/react-native-firebase/blob/v11.5.0/docs/admob/european-user-consent.md

**Reply by jeremyhalin** [reply]
Hi, thanks for this answer. 
I know it was possible with AdMob specifically, but the thing I'm talking about is a tool for developers by Google to get consent about user data. Analytics for example also collect data and we need to have user's consent.
The Google's tool is super effective and easy to implement, would be amazing to have in `react-native-firebase`.

For now, I'm collecting user consent in a manual manner.

**Reply by mikehardy** [reply]
Interesting - there is some separate discussion with the repo owner for react-native-ad-consent now that we have the AdMob repo alive and running post-extraction from react-native-firebase. The goal of that discussion is to bring the UMP features to the new google mobile ads repo and have it well maintained there.

I see what you mean about collecting data for non-ad-related items (e.g., analytics) but the UMP platform will not be useful for that I don't think - it has requirements that are all ad-related (you have to have an Ad Manager account with a funding choices account etc etc https://developers.google.com/ad-manager/ump/ios/quick-start - as such I do not think it makes sense to expose those APIs outside of a google mobile ads repository context. Stated differently: UMP does not make sense in react-native-firebase, it fits perfectly in google mobile ads repo though.

For ATT compliance with analytics or crashlytics or performance monitoring etc I believe there is no other option but to continue as you are now (and as I do in my work projects...) - collecting consent manually, as it makes sense for you to do in whatever legal jurisdiction you're in
